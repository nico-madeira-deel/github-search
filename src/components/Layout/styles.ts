import styled, { css, DefaultTheme } from 'styled-components'
import { LayoutProps } from '.'

const modifiers = {
  condensed: {
    wrapper: (theme: DefaultTheme) => css`
      height: auto;

      & > div {
        display: flex;
        justify-content: space-between;
      }

      header {
        flex-direction: row;

        h1 {
          font-size: ${theme.font.sizes.large};
          margin-left: ${theme.spacings.xsmall};
        }
      }

      .row {
        display: flex;
        align-items: flex-end;
        justify-content: center;

        button {
          margin-left: ${theme.spacings.xxsmall};
        }
      }
    `
  }
}

export const Main = styled.main`
  width: 100%;
`

export const Wrapper = styled.div<LayoutProps>`
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  max-width: 500px;
  margin: 0 auto;
  text-align: center;
  width: 100%;

  ${({ condensed, theme }) => css`
    transition: all ${theme.transitions.fast};
    padding: 0 ${theme.spacings.xsmall};
    padding-top: ${theme.spacings.xsmall};
    ${condensed && modifiers.condensed.wrapper(theme)}
  `}
`

export const Header = styled.header`
  align-items: center;
  display: flex;
  flex-direction: column;
`

export const Title = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: ${theme.font.bold};
    margin: ${theme.spacings.small} 0;
  `}
`

export const Description = styled.p`
  margin: 0;
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
  `}
`
