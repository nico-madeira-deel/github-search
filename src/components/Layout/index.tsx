import { useState } from 'react'
import * as S from './styles'
import Logo from 'components/Logo'
import Input from 'components/Input'
import Button from 'components/Button'
import { useUser } from 'contexts/UserContext'

export type LayoutProps = {
  children?: React.ReactNode
  condensed?: boolean
}

const Layout = ({ children, condensed = false }: LayoutProps) => {
  const [buttonDisabled, setButtonDisabled] = useState(true)
  const [searchValue, setSearchValue] = useState('')
  const { fetchUserData } = useUser()

  const handleInputChange = (value: string) => {
    setButtonDisabled(!value)
  }

  const handleInputBlur = (value: string) => {
    if (value) {
      setSearchValue(value)
    }
  }

  return (
    <S.Wrapper data-testid="layout" condensed={condensed}>
      <S.Header>
        <Logo size={condensed ? 'small' : 'normal'} />
        <S.Title>Github Search</S.Title>
      </S.Header>
      <S.Main>
        {!condensed ? (
          <S.Description>
            An application for search users and users repositories in github
          </S.Description>
        ) : null}
        <div className="row">
          <Input
            callbackOnBlur={(value) => handleInputBlur(value)}
            callbackOnChange={(value) => handleInputChange(value)}
            size={condensed ? 'small' : 'normal'}
            spaceBetween={!condensed}
          />
          <Button
            disabled={buttonDisabled}
            onClick={() => fetchUserData(searchValue)}
          >
            Enter
          </Button>
        </div>
        {children}
      </S.Main>
    </S.Wrapper>
  )
}

export default Layout
