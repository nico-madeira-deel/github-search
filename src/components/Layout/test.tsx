import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Layout from '.'

describe('<Layout />', () => {
  it('should render the layout properly', () => {
    const { container } = renderWithTheme(<Layout />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the layout condensed when required', () => {
    const { container } = renderWithTheme(<Layout condensed={true} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should have smaller height when condensed', () => {
    renderWithTheme(<Layout condensed={true} />)

    expect(screen.getByTestId('layout')).toHaveStyle({ height: 'auto' })
  })
})
