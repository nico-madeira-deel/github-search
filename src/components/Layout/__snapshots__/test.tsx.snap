// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the layout condensed when required 1`] = `
.c4 {
  width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 500px;
  margin: 0 auto;
  text-align: center;
  width: 100%;
  -webkit-transition: all .3s ease-in-out;
  transition: all .3s ease-in-out;
  padding: 0 1.6rem;
  padding-top: 1.6rem;
  height: auto;
}

.c0 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 header {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 header h1 {
  font-size: 1.8rem;
  margin-left: 1.6rem;
}

.c0 .row {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .row button {
  margin-left: 0.8rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  color: #faf9f9;
  font-size: 3.0rem;
  font-weight: 700;
  margin: 2.4rem 0;
}

.c2 {
  color: #faf9f9;
  height: auto;
  width: 3rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  width: 100%;
}

.c7 {
  background-color: transparent;
  border: 0.2rem solid #faf9f9;
  border-radius: 0.5rem;
  color: #faf9f9;
  padding-left: 0.8rem;
  -webkit-transition: border-width .5s ease-in-out;
  transition: border-width .5s ease-in-out;
  outline: none;
  height: 3.4rem;
  font-size: 1.6rem;
}

.c7:focus {
  border-width: 0.4rem;
}

.c6 {
  color: #faf9f9;
  font-size: 1.6rem;
  margin-bottom: 0.8rem;
}

.c8 {
  border: none;
  cursor: pointer;
  outline: none;
  background-color: #ffd6ba;
  border-radius: 0.5rem;
  color: #555b6e;
  font-size: 1.6rem;
  font-weight: 700;
  padding: 0.8rem 2.4rem;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c8:hover,
.c8:focus {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c8:disabled {
  cursor: not-allowed;
  opacity: 0.5;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<div
  class="c0"
  data-testid="layout"
>
  <header
    class="c1"
  >
    <div
      class="c2"
      color="white"
    >
      <svg
        aria-label="Github"
        fill="none"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 .5C5.37.5 0 5.78 0 12.292c0 5.211 3.438 9.63 8.205 11.188.6.111.82-.254.82-.567 0-.28-.01-1.022-.015-2.005-3.338.711-4.042-1.582-4.042-1.582-.546-1.361-1.335-1.725-1.335-1.725-1.087-.731.084-.716.084-.716 1.205.082 1.838 1.215 1.838 1.215 1.07 1.803 2.809 1.282 3.495.981.108-.763.417-1.282.76-1.577-2.665-.295-5.466-1.309-5.466-5.827 0-1.287.465-2.339 1.235-3.164-.135-.298-.54-1.497.105-3.121 0 0 1.005-.316 3.3 1.209.96-.262 1.98-.392 3-.398 1.02.006 2.04.136 3 .398 2.28-1.525 3.285-1.209 3.285-1.209.645 1.624.24 2.823.12 3.121.765.825 1.23 1.877 1.23 3.164 0 4.53-2.805 5.527-5.475 5.817.42.354.81 1.077.81 2.182 0 1.578-.015 2.846-.015 3.229 0 .309.21.678.825.56C20.565 21.917 24 17.495 24 12.292 24 5.78 18.627.5 12 .5z"
          fill="currentColor"
        />
      </svg>
    </div>
    <h1
      class="c3"
    >
      Github Search
    </h1>
  </header>
  <main
    class="c4"
  >
    <div
      class="row"
    >
      <div
        class="c5"
      >
        <label
          class="c6"
          for="search"
        >
          Search
        </label>
        <input
          aria-label="search"
          class="c7"
          id="search"
          name="search"
          type="text"
          value=""
        />
      </div>
      <button
        class="c8"
        disabled=""
        type="button"
      >
        Enter
      </button>
    </div>
  </main>
</div>
`

exports[`<Layout /> should render the layout properly 1`] = `
.c4 {
  width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 500px;
  margin: 0 auto;
  text-align: center;
  width: 100%;
  -webkit-transition: all .3s ease-in-out;
  transition: all .3s ease-in-out;
  padding: 0 1.6rem;
  padding-top: 1.6rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  color: #faf9f9;
  font-size: 3.0rem;
  font-weight: 700;
  margin: 2.4rem 0;
}

.c5 {
  margin: 0;
  color: #faf9f9;
  font-size: 1.6rem;
}

.c2 {
  color: #faf9f9;
  height: auto;
  width: 5rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  width: 100%;
  margin: 2.4rem 0;
}

.c8 {
  background-color: transparent;
  border: 0.2rem solid #faf9f9;
  border-radius: 0.5rem;
  color: #faf9f9;
  padding-left: 0.8rem;
  -webkit-transition: border-width .5s ease-in-out;
  transition: border-width .5s ease-in-out;
  outline: none;
  height: 5rem;
  font-size: 1.8rem;
}

.c8:focus {
  border-width: 0.4rem;
}

.c7 {
  color: #faf9f9;
  font-size: 1.8rem;
  margin-bottom: 0.8rem;
}

.c9 {
  border: none;
  cursor: pointer;
  outline: none;
  background-color: #ffd6ba;
  border-radius: 0.5rem;
  color: #555b6e;
  font-size: 1.6rem;
  font-weight: 700;
  padding: 0.8rem 2.4rem;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c9:hover,
.c9:focus {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c9:disabled {
  cursor: not-allowed;
  opacity: 0.5;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<div
  class="c0"
  data-testid="layout"
>
  <header
    class="c1"
  >
    <div
      class="c2"
      color="white"
    >
      <svg
        aria-label="Github"
        fill="none"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 .5C5.37.5 0 5.78 0 12.292c0 5.211 3.438 9.63 8.205 11.188.6.111.82-.254.82-.567 0-.28-.01-1.022-.015-2.005-3.338.711-4.042-1.582-4.042-1.582-.546-1.361-1.335-1.725-1.335-1.725-1.087-.731.084-.716.084-.716 1.205.082 1.838 1.215 1.838 1.215 1.07 1.803 2.809 1.282 3.495.981.108-.763.417-1.282.76-1.577-2.665-.295-5.466-1.309-5.466-5.827 0-1.287.465-2.339 1.235-3.164-.135-.298-.54-1.497.105-3.121 0 0 1.005-.316 3.3 1.209.96-.262 1.98-.392 3-.398 1.02.006 2.04.136 3 .398 2.28-1.525 3.285-1.209 3.285-1.209.645 1.624.24 2.823.12 3.121.765.825 1.23 1.877 1.23 3.164 0 4.53-2.805 5.527-5.475 5.817.42.354.81 1.077.81 2.182 0 1.578-.015 2.846-.015 3.229 0 .309.21.678.825.56C20.565 21.917 24 17.495 24 12.292 24 5.78 18.627.5 12 .5z"
          fill="currentColor"
        />
      </svg>
    </div>
    <h1
      class="c3"
    >
      Github Search
    </h1>
  </header>
  <main
    class="c4"
  >
    <p
      class="c5"
    >
      An application for search users and users repositories in github
    </p>
    <div
      class="row"
    >
      <div
        class="c6"
      >
        <label
          class="c7"
          for="search"
        >
          Search
        </label>
        <input
          aria-label="search"
          class="c8"
          id="search"
          name="search"
          type="text"
          value=""
        />
      </div>
      <button
        class="c9"
        disabled=""
        type="button"
      >
        Enter
      </button>
    </div>
  </main>
</div>
`
